src/sagemaker/jumpstart/curated_hub/hub_model_specs/hub_model_specs.py

```diff
@@ -0,0 +1,241 @@
+    """Frameowrk image for Private Hub model"""
+
+    Framework: str
+    FrameworkVersion: str
+    PythonVersion: str
+    TransformersVersion: Optional[str]
+    BaseFramework: Optional[str]
+
+
+@dataclass
+class ModelArtifactConfig:
+    """Model artifact location for Private Hub model"""
+
+    ArtifactLocation: str  # s3 address
+
+
+@dataclass
+class ScriptConfig:
+    """Script location for Private Hub model"""
+
+    ScriptLocation: str  # s3 address
+
+
+@dataclass
+class InstanceConfig:
+    """Instance configuration for Private Hub model"""
+
+    DefaultInstanceType: str
+    InstanceTypeOptions: List[str]
+
+
+@dataclass
+class Hyperparameter:
+    """Hyperparameter for Private Hub model"""
+
+    Name: str
+    DefaultValue: Optional[str]
+    Type: str  # enum
+    Options: List[Optional[str]]
+    Label: Optional[str]
+    Description: Optional[str]
+    Regex: Optional[str]
+    Min: Optional[str]
+    Max: Optional[str]
+
+
+@dataclass
+class ExtraChannels:
+    """Extra channels for Private Hub model"""
+
+    ChannelName: str
+    ChannelDataLocation: str  # s3 uri (prefix or key)
+
+
+@dataclass
+class DefaultTrainingConfig:
+    """Default training configuration arguments for Private Hub model"""
+
+    SdkArgs: Optional[DefaultTrainingSdkArgs]
+    CustomImageConfig: Optional[CustomImageConfig]
+    FrameworkImageConfig: Optional[FrameworkImageConfig]
+    ModelArtifactConfig: Optional[ModelArtifactConfig]
+    ScriptConfig: Optional[ScriptConfig]
+    InstanceConfig: Optional[InstanceConfig]
+    Hyperparameters: List[Hyperparameter]
+    ExtraChannels: List[ExtraChannels]
+
+
+@dataclass
+class InferenceNotebookConfig:
+    """Jupyter notebook location for Private Hub model"""
+
+    NotebookLocation: str  # s3 uri,
+
+
+@dataclass
+class DefaultDeploymentConfig:
+    """Default deployment configuration arguments for Private Hub model"""
+
+    SdkArgs: Optional[DefaultDeploymentSdkArgs]
+    CustomImageConfig: Optional[CustomImageConfig]
+    FrameworkImageConfig: Optional[FrameworkImageConfig]
+    ModelArtifactConfig: Optional[ModelArtifactConfig]
+    ScriptConfig: Optional[ScriptConfig]
+    InstanceConfig: Optional[InstanceConfig]
+    InferenceNotebookConfig: Optional[InferenceNotebookConfig]
+
+
+class ModelCapabilities(str, Enum):
+    """Model capabilities of a Private Hub model"""
+
+    TRAINING = "Training"
+    INCREMENTAL_TRAINING = "IncrementalTraining"
+    VALIDATION = "Validation"
+
+
+@dataclass
+class HubModelSpec_v1_0_0:
+    """Dataclass for Private Hub Model document"""
+
+    Capabilities: List[ModelCapabilities]  # enum?
+    DataType: str
+    MlTask: str
+    Framework: Optional[str]
+    Origin: Optional[Origin]
+    Dependencies: List[Dependency]
+    DatasetConfig: Optional[DatasetConfig]
+    DefaultTrainingConfig: Optional[DefaultTrainingConfig]
+    DefaultDeploymentConfig: Optional[DefaultDeploymentConfig]
+
+
+def convert_public_model_hyperparameter_to_hub_hyperparameter(
```

Comment:
can we move this to another module? this file seems to only have classes/enums, not utility functions

Reply:
+1